nifi-nar-bundles/nifi-asn1-bundle/nifi-asn1-services/src/main/java/org/apache/nifi/jasn1/convert/converters/TbcdStringConverter.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.jasn1.convert.converters;
+
+import com.beanit.asn1bean.ber.types.BerOctetString;
+import com.beanit.asn1bean.ber.types.BerType;
+import org.apache.nifi.jasn1.convert.JASN1Converter;
+import org.apache.nifi.jasn1.convert.JASN1TypeAndValueConverter;
+import org.apache.nifi.serialization.record.DataType;
+import org.apache.nifi.serialization.record.RecordFieldType;
+
+public class TbcdStringConverter implements JASN1TypeAndValueConverter {
+
+    private static final String TBCD_STRING_TYPE = "TBCDSTRING";
+    private static final char[] TBCD_SYMBOLS = "0123456789*#abc".toCharArray();
+
+    @Override
+    public boolean supportsType(Class<?> berType) {
+        boolean supportsType = BerOctetString.class.isAssignableFrom(berType) && isTbcdString(berType);
+
+        return supportsType;
+    }
+
+    @Override
+    public DataType convertType(Class<?> berType, JASN1Converter converter) {
+        DataType dataType = RecordFieldType.STRING.getDataType();
+
+        return dataType;
+    }
+
+    @Override
+    public boolean supportsValue(BerType value, DataType dataType) {
+        boolean supportsValue = value instanceof BerOctetString && isTbcdString(value.getClass());
+
+        return supportsValue;
+    }
+
+    @Override
+    public Object convertValue(BerType value, DataType dataType, JASN1Converter converter) {
+        final BerOctetString berValue = ((BerOctetString) value);
+
+        byte[] bytes = berValue.value;
+
+        int size = (bytes == null ? 0 : bytes.length);
+        StringBuilder resultBuilder = new StringBuilder(2 * size);
+
+        for (int octetIndex = 0; octetIndex < size; ++octetIndex) {
+            int octet = bytes[octetIndex];
+
+            int digit2 = (octet >> 4) & 0xF;
+            int digit1 = octet & 0xF;
+
+            if (digit1 == 15) {
```

Comment:
It would be helpful to define a `private static final` value for `15`, perhaps named something `MAXIMUM_DECIMAL_CODE`?

Reply:
Hmm, it's a bit confusing. But the point here is not that this is the max value, but that it's specifically 15, which is a "filler" value (if I understand correctly).
I can add such a constant.

Reply:
Thanks, for the reply, I was also unsure of what to call the constant, so I'm open to what makes sense. Perhaps `FILLER_DECIMAL_CODE`?